code/webauthn/frontend/src/pages/mint.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { Provider } from 'zksync-ethers';
+import { getDataToSign, signAndSend, authenticate } from '../../utils/webauthn';
+import { getTransaction } from '../../utils/tx';
+import React, { useEffect, useState } from 'react';
+import { Layout } from '../components/Layout';
+import { buttonStyles } from '.';
+import { containerStyles } from './register';
+import { ethers } from 'ethers';
+import NFT_ABI_JSON from '../../../contracts/artifacts-zk/contracts/MyNFT.sol/MyNFT.json';
+import { BUTTON_COLORS, NFT_CONTRACT_ADDRESS } from '@/pages/index';
+import { useAccount } from '@/hooks/useAccount';
+
+export default function Mint({ provider }: { provider: Provider }) {
+  const [isMounted, setIsMounted] = useState<boolean>(false);
+  const [mintedSVG, setMintedSVG] = React.useState<string | null>(null);
+
+  const account = useAccount();
+
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
+
+  async function mint(e: any) {
+    e.preventDefault();
+    try {
+      const contract = new ethers.Contract(NFT_CONTRACT_ADDRESS, NFT_ABI_JSON.abi, provider);
+      const functionName = 'mintZeek';
+      const functionArgs: any[] = [];
+      const data = contract.interface.encodeFunctionData(functionName, functionArgs);
+      const transferValue = '0';
+      const tx = await getTransaction(NFT_CONTRACT_ADDRESS, account!, transferValue, data, provider);
+      const signedTxHash = getDataToSign(tx);
+      const authResponse = await authenticate(signedTxHash.toString());
+      const receipt = await signAndSend(provider, tx, authResponse);
```

Comment:
I'm a little surprised you are building the transaction signature by hand instead of using the SmartAccount helper classes, but honestly this might be more clear and this isn't supposed to be an example of how to use a SmartAccount specifically.

Reply:
interesting, it seems like a private key is required to use that though? In this component no private key is required. It might make sense to use that in the register component though 👀

Reply:
Not really, you can use some 'secret' but it doesn't need to be a private key!
https://github.com/matter-labs/zksync-account-sdk/blob/main/packages/contracts/test/EndToEndSpendLimit.ts#L301

I don't know how important this is.